### PR TITLE
fix(redshift): allow standard_conforming_strings option

### PR DESCRIPTION
**Pull request by aheuermann**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

We use sequelize with Redshift using the sequelize postgres dialect. It doesn't support the `standard_conforming_strings` option since it is based on an older version of postgres.  Since https://github.com/sequelize/sequelize/commit/850c7fd04669e0fef9238b6dc4f8d6ee93ed71e9 was introduced,  sequelize will try to set this option and fail, so I added a config option.

### CODE REVIEW

File: types/lib/sequelize.d.ts [hunk 1]
```diff
@@ -305,6 +305,15 @@ export interface Options extends Logging {
    */
   operatorsAliases?: OperatorsAliases | false;
 
+
+  /**
+   * The PostgreSQL `standard_conforming_strings` session parameter. Set to `false` to not set the option.
+   * WARNING: Setting this to false may expose vulnerabilities and is not recommended!
+   *
+   * @default all aliases
```
Comment (SimonSchick) [inline types/lib/sequelize.d.ts]: Nit: Should be `true`.

File: types/lib/sequelize.d.ts [hunk 2]
```diff
@@ -305,6 +305,15 @@ export interface Options extends Logging {
    */
   operatorsAliases?: OperatorsAliases | false;
```
Comment (SimonSchick) [inline types/lib/sequelize.d.ts]: Optional: While we are here, can you remove `| false` here pretty please? :)

**Comment by aheuermann** [comment]
Fixed, thanks for taking a look!

I removed the tests as they weren't really testing anything since the  default is `on` and the query to set the option runs on connect and is per-session. I couldn't find a good way to spy on the query itself. If you have any suggestions let me know, and I can add a test.

**Comment by sushantdhiman** [comment]
This should be support by different `dialect = redshift` or version check, it can leave users open to SQL injection if redshift updated their postgres version

**Pull request merged**